core/tee/tee_svc.c

```diff
@@ -56,11 +56,11 @@ void syscall_log(const void *buf __maybe_unused, size_t len __maybe_unused)
 	kbuf = malloc(len);
 	if (kbuf == NULL)
 		return;
-	*kbuf = '\0';
 
-	/* log as Info/Raw traces */
-	if (tee_svc_copy_from_user(kbuf, buf, len) == TEE_SUCCESS)
-		TAMSG_RAW("%.*s", (int)len, kbuf);
+	if (tee_svc_copy_from_user(kbuf, buf, len) == TEE_SUCCESS) {
+		kbuf[len] = '\0';
+		trace_ext_puts(kbuf);
```

Comment:
TA can straight call the trace syscall to bypass libutee prefix. Actually, you only need to use `printf()` to do so. If you feel it's up to the TA developer to conform with `xMSG()` to get trace prefixing, then I am ok.

When the same console (uart?) mixes nonsecure trace and secure traces, it's nice to identify traces. Yet this deep thought does not conflict with discarding 'tee core' prefixing from TA traces.

I agree current prefixing is too verbose and should be lighted.

> So, sorry to say Etienne (...)

don't be sorry. i prefer stating my feeling and get people feedback (not to say that most of the time, i am wrong).